File: schedule-service/src/main/java/pe/albrugroup/schedule_service/entity/response/horario/AmpliacionContextoResponse.java
package pe.albrugroup.schedule_service.entity.response.horario;

import lombok.*;
import pe.albrugroup.schedule_service.entity.enums.EstadoAsistencia;

import java.time.LocalDate;
import java.time.LocalTime;

/**
 * Contexto que necesita la UI para plantear una ampliacion de un empleado en una fecha,
 * sin pedir IDs tecnicos ni requerir READ_HORARIOS.
 */
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AmpliacionContextoResponse {

    private Long idEmpleado;
    private LocalDate fecha;

    private Boolean tieneHorarioVigente;
    private Boolean laborable;

    /** Ventana vigente efectiva del dia (base + ampliaciones ya aplicadas). */
    private LocalTime entradaProgramada;
    private LocalTime salidaProgramada;

    private EstadoAsistencia estadoActual;
    /** El empleado ya marco ingreso hoy. */
    private Boolean tieneRegistro;
    /** La jornada del dia ya fue cerrada (marco salida). Habilita el flujo de reapertura. */
    private Boolean jornadaCerrada;
}

File: frontend/src/app/shared/models/schedule/ampliacion-contexto-response.ts
import { EstadoAsistencia } from './estado-asistencia';

/** Contexto del horario de un empleado en una fecha, para plantear una ampliacion. */
export interface AmpliacionContextoResponse {
  idEmpleado: number;
  fecha: string;
  tieneHorarioVigente: boolean;
  laborable: boolean;
  /** "HH:mm:ss" o null. */
  entradaProgramada: string | null;
  salidaProgramada: string | null;
  estadoActual: EstadoAsistencia;
  tieneRegistro: boolean;
  jornadaCerrada: boolean;
}
